client/client_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"database/sql"
+	"io/ioutil"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/google/trillian"
+	"github.com/google/trillian/extension/builtin"
+	"github.com/google/trillian/server"
+	"github.com/google/trillian/util"
+	"google.golang.org/grpc"
+)
+
+const (
+	createSQLFile = "../storage/mysql/storage.sql"
+	mysqlURI      = "root@tcp(127.0.0.1:3306)/"
+)
+
+var mysqlURIdb = "root@tcp(127.0.0.1:3306)/unittest"
+
+type Env struct {
+	grpcServer *grpc.Server
+	logServer  *server.TrillianLogRPCServer
+	Client     *LogClient
+}
+
+// Listen opens a random high numbered port for listening.
+func Listen() (string, net.Listener, error) {
+	lis, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return "", nil, err
+	}
+	_, port, err := net.SplitHostPort(lis.Addr().String())
+	if err != nil {
+		return "", nil, err
+	}
+	addr := "localhost:" + port
+	return addr, lis, nil
+}
+
+// RestDB drops and recreates the test database.
+// TODO: convert to CreateDB when registry can take a db object.
+func ResetDB() error {
+	builtin.MySQLURIFlag = &mysqlURIdb
+
+	db, err := sql.Open("mysql", mysqlURI)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	resetSQL := []string{
+		"DROP DATABASE IF EXISTS unittest;",
+		"CREATE DATABASE unittest;",
+		"GRANT ALL ON unittest.* TO 'unittest'@'localhost' IDENTIFIED BY 'zaphod';",
+	}
+	for _, sql := range resetSQL {
+		if _, err := db.Exec(sql); err != nil {
+			return err
+		}
+	}
+
+	// Reconnect to use the new test database.
+	dbTest, err := sql.Open("mysql", mysqlURIdb)
+	if err != nil {
+		return err
+	}
+	defer dbTest.Close()
+
+	createSQL, err := ioutil.ReadFile(createSQLFile)
+	if err != nil {
+		return err
+	}
+
+	sqlSlice := strings.Split(string(createSQL), ";\n")
```

Comment:
I'm worried this is brittle -- I think it will go wrong if anyone 
 - leaves a trailing space after a semicolon
 - ends a comment with a semicolon
 - drops the final newline
in `storage.sql`.